actix-session/src/lib.rs

```diff
@@ -332,6 +332,13 @@ pub mod test_helpers {
                     .service(resource("/do_something").route(post().to(do_something)))
                     .service(resource("/login").route(post().to(login)))
                     .service(resource("/logout").route(post().to(logout)))
+                    .service(
+                        web::scope("/protected")
```

Comment:
This test is already fairly complicated as it is, considering the complexity of the scenario we are going through - I'd prefer not to add more stuff to it.
Can we write separate tests that exercises the guard method? One for the "happy case" (there is a session, we can extract it successfully), one for the unhappy case (there is no session cookie, the request is rejected/fails).